src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -313,6 +313,14 @@ public BoundExpression Property(BoundExpression? receiverOpt, PropertySymbol pro
             return Call(receiverOpt, accessor);
         }
 
+        public BoundExpression Indexer(BoundExpression? receiverOpt, PropertySymbol property, BoundExpression arg0)
```

Comment:
I had a feeling list-patterns might be using this overload but my "delete unused method" impulse won out.